### PR TITLE
TF-2384 [PART-2] Improvement notification iOS

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#2384 

## Task

- [x] Show notification with list emails
- [x] Handle refresh token when token expired

### CODE REVIEW

File: ios/TwakeMailNSE/NotificationService.swift [hunk 1]
```diff
@@ -1,30 +1,33 @@
 import UserNotifications
 
 class NotificationService: UNNotificationServiceExtension {
-    
+
+    private let timeIntervalNotificationTrigger: Int = 2
```
Comment (hoangdat) [inline ios/TwakeMailNSE/NotificationService.swift:5]: please specific the time unit: `timeIntervalNotificationTriggerInSecond`

File: ios/TwakeCore/Network/Interceptor/AuthenticationInterceptor.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+            self.keychainController.updateTokenOidc(
+                accountId: self.accountId,
+                newTokenOidc: TokenOidc(
+                    token: accessToken,
+                    tokenId: tokenResponse.tokenId,
+                    expiredTime: "\(tokenResponse.expiresTime ?? 0)",
+                    refreshToken: refreshToken
+                )
+            )
```
Comment (hoangdat) [inline ios/TwakeCore/Network/Interceptor/AuthenticationInterceptor.swift:59]: Flutter side can use this updated token or not?

Comment (hoangdat) [inline ios/TwakeCore/Network/Interceptor/AuthenticationInterceptor.swift:59]: What happens if `NSE refreshed token` then `Flutter side` also refresh token?

Comment (dab246) [reply ios/TwakeCore/Network/Interceptor/AuthenticationInterceptor.swift:59]: We will handle it in the interceptor. When a 401 error response is returned, we will use the newly updated `token` in the keychain to continue making the request. If not, a `refreshToken` request will be made

**Review by hoangdat — APPROVED** [review]

**Pull request merged**